cobalt/precommit/check_java_strings_have_locale.py

```diff
@@ -0,0 +1,33 @@
+import argparse
+import re
+import sys
+
+
+def main():
+  parser = argparse.ArgumentParser()
+  parser.add_argument('files', nargs='*')
+  args = parser.parse_args()
+
+  errors = []
+  # This regex looks for String.format( where the rest of the line does not
+  # contain "Locale". This is a heuristic for single-line calls.
+  pattern = re.compile(r'\bString\.format\s*\((?!.*Locale)')
+
+  for file_path in args.files:
+    try:
+      with open(file_path, 'r', encoding='utf-8') as f:
+        for line_num, line in enumerate(f, 1):
+          if pattern.search(line):
+            errors.append(f'{file_path}:{line_num}: String.format call '
+                          'should include a java.util.Locale argument.')
+    except Exception as e:
+      print(f"Error processing file {file_path}: {e}", file=sys.stderr)
```

Comment:
You want to catch any possible error and succeed? Why?

Reply:
I think the rationale in these filters is to only catch one thing,
in this case the string-w/o-locale; if the files fail to open for
some reason, the presubmit won't hold the commit. Makes sense? 
Otherwise LMK and I can append an `error` here as well.

Reply:
Gonna resolve this convo to submit this CL/PR. @andrewsavage1 if you
still have concerns we can address them in subsequent CL/PRs.